curp/src/server/cmd_worker/mod.rs

```diff
@@ -174,6 +174,33 @@ async fn worker_as<C: Command, CE: CommandExecutor<C>, RC: RoleChange>(
             sp.lock().remove(&entry.propose_id);
             let _ig = ucp.lock().remove(&entry.propose_id);
             if shutdown_self {
+                if let Some(maybe_new_leader) = curp.pick_new_leader() {
```

Comment:
Is the else branch unreachable? We could also add a log to the else branch.

Reply:
Added